frontend/src/metabase/components/SaveQuestionForm/util.ts

```diff
@@ -24,38 +28,53 @@ const updateQuestion = async (
   await onSave(updatedQuestion.setId(originalQuestion.id()));
 };
 
-export const createQuestion = async (
-  details: FormValues,
-  question: Question,
-  onCreate: (question: Question) => Promise<void>,
-) => {
+export const createQuestion = async (options: CreateQuestionOptions) => {
+  const { details, question, onCreate, saveToCollectionId } = options;
+
   if (details.saveType !== "create") {
     return;
   }
 
-  const collectionId = canonicalCollectionId(details.collection_id);
+  // `saveToCollectionId` is used to override the target collection of the question,
+  // this is mainly used for the embedding sdk.
+  const collectionId = isNullOrUndefined(saveToCollectionId)
```

Comment:
I'd love to see you keep `canonicalCollectionId` here since you keep the variable name the same, but you move `canonicalCollectionId` somewhere else.

This feels like the meaning of this variable already changed, then we might need to change the name.

The nuance of `collectionId` in this version isn't a canonical collection ID anymore.

Reply:
cleaned up in https://github.com/metabase/metabase/pull/49251/commits/c60cf9a62798a62bb0b734580b5fd5baac01f134